### PR TITLE
Export files refactor

### PR DESCRIPTION
**Pull request by chr1sgrant**

### CODE REVIEW

File: antha/AnthaStandardLibrary/Packages/export/export.go [hunk 1]
```diff
@@ -98,168 +101,243 @@ func Exporttofile(dir string, seq wtype.BioSequence) (string, error) {
...
+	anthafile.WriteAll(allbytes)
 
-	return filename, nil
+	return anthafile, filename, err
 }
 
-func Makefastaserial2(makeinanthapath bool, dir string, seqs []wtype.DNASequence) (string, error) {
+// function to export multiple sequences in fasta format into a specified directory
+// specify whether to save locally or to the anthapath in a specified sub directory dir.
+func FastaSerial(makeinanthapath bool, dir string, seqs []wtype.DNASequence) (wtype.File, string, error) {
```
Comment (jarpinosynthace) [inline antha/AnthaStandardLibrary/Packages/export/export.go]: does FastaSerial generate a single file per sequence or does it generate a single file containing many sequences. Maybe clarify this in the description. Also does the function name FastaSerial then describe this properly, FastaMultiFile or something like that maybe?

File: antha/AnthaStandardLibrary/Packages/export/export.go [hunk 2]
```diff
@@ -98,168 +101,243 @@ func Exporttofile(dir string, seq wtype.BioSequence) (string, error) {
...
 
 	}
 
-	return Makefastaserial2(ANTHAPATH, dirname, seqs)
+	return FastaSerial(ANTHAPATH, dirname, seqs)
 }
 
-func ExporttoTextFile(filename string, data []string) error {
+// export data in the format of an array of strings to a file
+func TextFile(filename string, data []string) (wtype.File, error) {
```
Comment (jarpinosynthace) [inline antha/AnthaStandardLibrary/Packages/export/export.go]: DataToTextFile as name suggestion?

File: antha/AnthaStandardLibrary/Packages/export/export.go [hunk 3]
```diff
@@ -98,168 +101,243 @@ func Exporttofile(dir string, seq wtype.BioSequence) (string, error) {
...
+
+	anthafile.WriteAll(alldata)
 
-	return nil
+	return anthafile, nil
 }
 
-func ExporttoJSON(data interface{}, filename string) (err error) {
+// Export any data as a json object in  a file
+func JSON(data interface{}, filename string) (anthafile wtype.File, err error) {
```
Comment (jarpinosynthace) [inline antha/AnthaStandardLibrary/Packages/export/export.go]: DataToJSON name change suggestion

File: antha/AnthaStandardLibrary/Packages/export/export.go [hunk 4]
```diff
@@ -98,168 +101,243 @@ func Exporttofile(dir string, seq wtype.BioSequence) (string, error) {
...
 	ioutil.WriteFile(filename, bytes, 0644)
-	return nil
+
+	anthafile.Name = filename
+	anthafile.WriteAll(bytes)
+	return anthafile, nil
+}
+
+// Export a 2D array of string data as a csv file
+func CSV(records [][]string, filename string) (wtype.File, error) {
```
Comment (jarpinosynthace) [inline antha/AnthaStandardLibrary/Packages/export/export.go]: ToCSV suggested name change

File: antha/AnthaStandardLibrary/Packages/export/export.go [hunk 5]
```diff
@@ -98,168 +101,243 @@ func Exporttofile(dir string, seq wtype.BioSequence) (string, error) {
...
+	}
+	return anthafile, nil
+}
+
+// export a stream into a file
+func Stream(stream io.Reader, filename string) (wtype.File, error) {
+	return Binary(streamToByte(stream), filename)
+}
+
+func stringsToBytes(data []string) []byte {
```
Comment (jarpinosynthace) [inline antha/AnthaStandardLibrary/Packages/export/export.go]: Add comments in code describing what this does

File: antha/AnthaStandardLibrary/Packages/export/export.go [hunk 6]
```diff
@@ -98,168 +101,243 @@ func Exporttofile(dir string, seq wtype.BioSequence) (string, error) {
...
+	for _, str := range data {
+		bts := []byte(str)
+		for i := range bts {
+			alldata = append(alldata, bts[i])
+		}
+	}
+	return alldata
+}
+
+func streamToByte(stream io.Reader) []byte {
```
Comment (jarpinosynthace) [inline antha/AnthaStandardLibrary/Packages/export/export.go]: Add comments in code describing what this does

**Comment by chr1sgrant** [comment]
@synthacebot test this please

File: antha/AnthaStandardLibrary/Packages/export/export.go (hunk 1 above)
Comment (chr1sgrant) [inline antha/AnthaStandardLibrary/Packages/export/export.go]: This generates a single file with multiple sequences in the same file. I'll updated the description

File: antha/AnthaStandardLibrary/Packages/export/export.go (hunk 5 above)
Comment (chr1sgrant) [inline antha/AnthaStandardLibrary/Packages/export/export.go]: The descriptions are there mostly for visibility in the godocs, which only show public functions: https://godoc.org/github.com/antha-lang/antha/antha/AnthaStandardLibrary/Packages/export

These are not public functions and are only intended as helpers in this package.

**Pull request merged**